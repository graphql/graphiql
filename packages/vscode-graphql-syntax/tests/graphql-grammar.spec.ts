import { describe, it, expect } from 'vitest';
import { tokenizeFile } from './utilities';

describe('source.graphql grammar', () => {
  const scope = 'source.graphql';

  it('should tokenize a simple query', async () => {
    const result = await tokenizeFile('fixtures/query.graphql', scope);
    expect(result).toMatchSnapshot();
  });
});
