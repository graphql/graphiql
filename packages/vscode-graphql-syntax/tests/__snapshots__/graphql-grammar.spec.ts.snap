// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`source.graphql grammar > should tokenize a simple query 1`] = `
query  | keyword.operation.graphql
       | 
simple | entity.name.function.graphql
       | meta.selectionset.graphql
{      | meta.selectionset.graphql punctuation.operation.graphql
       | meta.selectionset.graphql
stuff  | meta.selectionset.graphql variable.graphql
(      | meta.selectionset.graphql meta.arguments.graphql meta.brace.round.directive.graphql
id     | meta.selectionset.graphql meta.arguments.graphql variable.parameter.graphql
:      | meta.selectionset.graphql meta.arguments.graphql punctuation.colon.graphql
       | meta.selectionset.graphql meta.arguments.graphql
"      | meta.selectionset.graphql meta.arguments.graphql string.quoted.double.graphql punctuation.definition.string.begin.graphql
1      | meta.selectionset.graphql meta.arguments.graphql string.quoted.double.graphql
"      | meta.selectionset.graphql meta.arguments.graphql string.quoted.double.graphql punctuation.definition.string.end.graphql
)      | meta.selectionset.graphql meta.arguments.graphql meta.brace.round.directive.graphql
       | meta.selectionset.graphql meta.selectionset.graphql
{      | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
       | meta.selectionset.graphql meta.selectionset.graphql
id     | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
       | meta.selectionset.graphql meta.selectionset.graphql
name   | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
       | meta.selectionset.graphql meta.selectionset.graphql
}      | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
}      | meta.selectionset.graphql punctuation.operation.graphql
       | 
`;
